config/application.php

```diff
@@ -52,10 +52,18 @@
 /**
  * DB settings
  */
-Config::define('DB_NAME', env('DB_NAME'));
-Config::define('DB_USER', env('DB_USER'));
-Config::define('DB_PASSWORD', env('DB_PASSWORD'));
-Config::define('DB_HOST', env('DB_HOST') ?: 'localhost');
+$hasDsn = env('DATABASE_URL') !== null ? true : false;
+
+if (true === $hasDsn) {
+    $databaseDsn = parse_url(env('DATABASE_URL'));
+    $dbName = substr($databaseDsn['path'], 1);
+    $dbHost = isset($databaseDsn['port']) ? $databaseDsn['host'] . ":" . $databaseDsn['port'] : $databaseDsn['host'];
+}
+
+Config::define('DB_NAME', $hasDsn ? $dbName : env('DB_NAME'));
```

Comment:
These `$hasDsn` ternary operations aren't needed. `Config::define()` can be declared more than once or you can set the env var.

In other words, you can use
```php
if (env('DB_URL')) {
    //...
    putenv('DB_...');// or $_ENV['DB_...'] = ...
    //...
}

Config::define('DB_...');
```

Or you could do the following...

```php
Config::define('DB_...');
// ...

if (env('DB_URL')) {
  Config::define('DB_...');
  //...
}
```

I personally prefer the latter.